Source/ACE/Network/GameMessages/Messages/GameMessageFellowshipFullUpdate.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using ACE.Common.Extensions;
+using ACE.Entity.Actions;
+using ACE.Entity;
+
+namespace ACE.Network.GameMessages.Messages
+{
+    public class GameMessageFellowshipFullUpdate : GameMessage
+    {
+        public GameMessageFellowshipFullUpdate(Session session)
+            : base(GameMessageOpcode.GameEvent, GameMessageGroup.Group09)
+        {
+            Fellowship fellowship = session.Player.Fellowship;
+            ActionChain chain = new ActionChain();
+            chain.AddAction(session.Player, () =>
+            {
+                Writer.Write(session.Player.Guid.Full);
+                Writer.Write(session.GameEventSequence++);
+                Writer.Write((uint)GameEvent.GameEventType.FellowshipFullUpdate);
+
+                // the current number of fellowship members
+                Writer.Write((UInt16)fellowship.FellowshipMembers.Count);
+
+                // ????
+                Writer.Write((byte)0x10);
+                Writer.Write((byte)0x00);
+
+                // --- FellowInfo ---
+                
+                foreach (Player fellow in fellowship.FellowshipMembers)
+                {
+                    if (fellow.Guid.Full != session.Player.Guid.Full)
+                    {
+                        chain.AddAction(fellow, () => { WriteFellow(fellow); });
```

Comment:
Correct me if I have any of this wrong.. `session.Player` is already locked from being added to the chain above, so we can't have any other actions try to read from the player. If we try to add `session.Player` as an Actor again, we won't be able to get any data from it in the action that is passed.

However, we still need to add the other `Player`s to the chain before we try to read from them, so the rest of the `Player`s are added to the chain.